### PR TITLE
Make gathering cancelable using abortable

### PR DESCRIPTION
**Pull request by matsui54**
Like [ddu-source-file_external](https://github.com/matsui54/ddu-source-file_external/blob/main/denops/%40ddu-sources/file_external.ts), by this PR, gathering is stopped properly when quit action is executed while gathering. Also, live grep is now working well.

live_grep

### CODE REVIEW

**Comment by Shougo** [comment]
> Also, live grep is now working well.

I think live grep way should be documented.

File: doc/ddu-source-rg.txt [hunk 1]
```diff
@@ -41,6 +41,24 @@ EXAMPLES					*ddu-source-rg-examples*
...
+    function! s:ddu_rg_live() abort
+      call ddu#start({
+            \ 'volatile': v:true,
+            \ 'sources': [{
+            \   'name': 'rg', 
+            \   'options': {'matchers': []},
+            \ }],
+            \ 'uiParams': {'ff': {
+            \   'ignoreEmpty': v:false,
+	    \   'autoResize': v:false,
```
Comment (Shougo) [inline doc/ddu-source-rg.txt:58]: The indentation seems wrong.

**Comment by Shougo** [comment]
It should be:

```diff
diff --git a/doc/ddu-source-rg.txt b/doc/ddu-source-rg.txt
index 69629e7..4bbcc3e 100644
--- a/doc/ddu-source-rg.txt
+++ b/doc/ddu-source-rg.txt
@@ -43,20 +43,20 @@ EXAMPLES                    *ddu-source-rg-examples*
     call ddu#start({'sources': [{'name': 'rg'}, 'params': {'input': word}]})

     " live grep
-    " You need to make volatile option v:true
-    " Note that matchers should be empty for performance
+    " You need to make "volatile" option v:true.
+    " Note: the matchers should be empty for performance
     command! DduRgLive call <SID>ddu_rg_live()
     function! s:ddu_rg_live() abort
       call ddu#start({
-            \ 'volatile': v:true,
-            \ 'sources': [{
-            \   'name': 'rg',
-            \   'options': {'matchers': []},
-            \ }],
-            \ 'uiParams': {'ff': {
-            \   'ignoreEmpty': v:false,
-       \   'autoResize': v:false,
-            \ }},
+            \   'volatile': v:true,
+            \   'sources': [{
+            \     'name': 'rg',
+            \     'options': {'matchers': []},
+            \   }],
+            \   'uiParams': {'ff': {
+            \     'ignoreEmpty': v:false,
+            \     'autoResize': v:false,
+            \   }},
             \ })
     endfunction
 <
```

**Comment by matsui54** [comment]
Thanks. Fixed.

**Comment by shun** [comment]
Thanks for your good PR !

**Pull request merged**